gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/bulk/BulkRequest.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.management.handlers.management.api.bulk;
+
+import io.gravitee.am.common.utils.Indexed;
+import io.reactivex.rxjava3.core.Single;
+import jakarta.ws.rs.core.Response;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.function.Function;
+
+public record BulkRequest<T>(List<T> items) {
```

Comment:
**question**: wondering if we should have a operation type in addition of the list of items. :thinking: to be able to make the distinction between creation or update bulk request.

Reply:
What I understood from discussions on slack/daily was there's going to be separate endpoints for different operations. So a "bulk create" endpoint would accept a BulkRequest<NewThing> and process accordingly, and a "bulk update" endpoint would accept a BulkRequest<UpdatedThing>.

If that's not the case, we'll need to change this class to force in-order execution of each "item".

Reply:
yes but which HTTP method do you plan to use ? With a step back single bulk action on UserResource maybe better but this is a discussion to have will all team member. 

But I don't understand why we'll need to change this class to force in-order execution of each "item".
I was thinking to the something like:

```
POST /organisation/DEFAULT/environment/DEFAULT/domain/domainId/users/bulk
Content-Type: application /json

{
  "action": "create", 
  "items": [ {"username":...} , ...]
}
```
or
```
POST /organisation/DEFAULT/environment/DEFAULT/domain/domainId/users/bulk
Content-Type: application /json

{
  "action": "update",
  "items": [ {"id":...} , ...]
}
```
or
```
POST /organisation/DEFAULT/environment/DEFAULT/domain/domainId/users/bulk
Content-Type: application /json

{
  "action": "delete",
  "items": [ {"id":...} , ...]
}
```

Reply:
oh, so still one action per request, just with a single endpoint handling each action? That makes sense, I'll update the request class accordingly